pkg/apis/pipeline/v1beta1/pipeline_types.go

```diff
@@ -170,6 +170,21 @@ type Matrix struct {
 	// The names of the `params` in the `Matrix` must match the names of the `params` in the underlying `Task` that they will be substituting.
 	// +listType=atomic
 	Params []Param `json:"params,omitempty"`
+
+	// Include is a list of MatrixInclude
+	// Note this field is in preview mode and not yet supported
+	Include []MatrixInclude `json:"include,omitempty"`
+}
+
+// MatrixInclude allows passing in a specific combinations of Parameters into the Matrix.
+// Note this struct is in preview mode and not yet supported
+type MatrixInclude struct {
```

Comment:
I thought this kind of struct might already be defined somewhere, but didn't find any 🙃